x/programs/rust/wasmlanche-sdk/src/state.rs

```diff
@@ -124,34 +128,41 @@ impl<'a, K: Key> State<'a, K> {
     pub fn delete<T: BorshDeserialize>(self, key: K) -> Result<Option<T>, Error> {
         #[link(wasm_import_module = "state")]
         extern "C" {
-            #[link_name = "delete"]
-            fn delete(ptr: *const u8, len: usize) -> HostPtr;
+            #[link_name = "put"]
+            fn put(ptr: *const u8, len: usize);
+        }
+
+        #[derive(BorshSerialize)]
+        struct PutArgs<Key> {
+            key: Key,
+            value: Vec<u8>,
+        }
+
+        let value = self.get(key)?;
+        if value.is_none() {
+            return Ok(None);
         }
 
         // TODO:
         // we should actually cache deletes as well
         // to avoid cache misses after delete
-        self.cache.borrow_mut().remove(&key);
-
-        let args_bytes = borsh::to_vec(&key).map_err(|_| StateError::Serialization)?;
+        let args = PutArgs {
+            key,
+            value: Vec::new(),
+        };
+        let args_bytes = borsh::to_vec(&vec![args]).map_err(|_| StateError::Serialization)?;
```

Comment:
```suggestion
        let args_bytes = borsh::to_vec(&[args]).map_err(|_| StateError::Serialization)?;
```

Reply:
added `as_slice` so it wasn't treated as a fixed length array by Borsh